src/trip-repo.js

```diff
@@ -3,51 +3,95 @@ import Trip from "./trip.js";
+
+    // const check = this.compareDates(this.allTrips.filter(element => {
+    //   if (this.compareDates(date, element.date)) {
+    //     console.log(element)
+    //     return element;
+    //   } 
+    // }));
+    
+    const pastTrips = [];
     
     this.compareDates(this.allTrips.forEach(element => {
       if (this.compareDates(date, element.date)) {
-        goodDates.push(element)
+        pastTrips.push(element);
       }
     }));
-    const currentUserTrips = goodDates.filter(trip => trip.userID === userID);
-    const destinationArray = this.matchDestinationNames(currentUserTrips);
-    destinationArray.forEach(element => {
-      dateDestinationArray.push({ 'date' : element.date, 'destination' : element.destination.destination })
-    });
-    return dateDestinationArray;
+
+    return this.createOnlyDateDestination(pastTrips, userID);
   }
 
-  matchDestinationNames(tripArray) {
-    const checkArray = [];
-    tripArray.forEach(element => {
-      const newDestination = this.destinationData.filter(dest => dest.id === element.id);
-      const newTrip = new Trip(element, newDestination[0]);
-      checkArray.push(newTrip);
-    });
-    return checkArray;
+  findUserCurrentTrip(userID, date) {
+    const currentUserTrips = this.allTrips.filter(trip => trip.date === date);
+    return this.createOnlyDateDestination(currentUserTrips, userID);
   }
 
-  // findUserCurrentTrip(userID, date) {
-  // }
+  findUserUpcomingTrips(userID, date) {
+    const futureTrips = [];
+    
+    this.compareDates(this.allTrips.forEach(element => {
+      if (this.compareDates(element.date, date)) {
+        futureTrips.push(element);
+      }
+    }));
+    return this.createOnlyDateDestination(futureTrips, userID);
+  }
 
-  // findUserUpcomingTrips(userID, date) {
-  // }
+  findUserPendingTrips(userID, date) {
+    const futureTrips = [];
+    
+    this.compareDates(this.allTrips.forEach(element => {
+      if (this.compareDates(element.date, date)) {
+        futureTrips.push(element);
+      }
+    }));
+    const pendingTrips = futureTrips.filter(trip => trip.status === 'pending');
+      return this.createOnlyDateDestination(pendingTrips, userID);
+  }
 
-  // findUserPendingTrips(userID) {
-  // }
+  calculateYearlyExpenditure(userID, date) {
+    const year = date.split('/')[0];
+    const pastDates = [];
+    let totalCost = 0;
+
+    this.compareDates(this.allTrips.forEach(element => {
+      if (this.compareDates(element.date, year) && (this.compareDates(date, element.date))) {
+        pastDates.push(element);
+      }
+    }));
+    const currentUserTrips = pastDates.filter(trip => trip.userID === userID);
+    const destinationArray = this.matchDestinationNames(currentUserTrips);    
+    destinationArray.forEach(trip => totalCost += trip.calculateCost());
```

Comment:
This is a great way to use `forEach` to add up the total cost! However, try to see if you can do it using `Array.reduce` instead with `0` as an accumulator.